supporting_scripts/generate_mermaid_erd.py

```
@@ -0,0 +1,617 @@
+                    WHERE table_schema = 'public' 
+                    AND table_type = 'BASE TABLE'
+                    AND table_name NOT IN ('databasechangelog', 'databasechangeloglock')
+                    ORDER BY table_name
+                """)
+                return [row[0] for row in cursor.fetchall()]
+        except Exception as e:
+            error_msg = f"Failed to fetch table names: {e}"
+            logger.error(error_msg)
+            raise DatabaseConnectionError(error_msg) from e
+
+    def get_table_columns(self, table_name: str) -> List[Dict]:
```

Pylint (3.3.7)

[refactor] Too many local variables (18/15)

(R0914)